prov/gni/src/gnix_av.c

```diff
@@ -446,12 +446,12 @@ static int map_insert(struct gnix_fid_av *int_av, const void *addr,
 
 		((struct gnix_address *)fi_addr)[i] = temp.gnix_addr;
 		the_entry =  &blk->base[i];
-		memcpy(&the_entry->gnix_addr, &temp.gnix_addr,
-		       sizeof(struct gnix_address));
+		the_entry->gnix_addr.device_addr = temp.gnix_addr.device_addr;
+		the_entry->gnix_addr.cdm_id = temp.gnix_addr.cdm_id;
 		the_entry->name_type = temp.name_type;
 		the_entry->cm_nic_cdm_id = temp.cm_nic_cdm_id;
 		the_entry->cookie = temp.cookie;
-		memcpy(&key, &temp.gnix_addr, sizeof(gnix_ht_key_t));
+		key = *((gnix_ht_key_t *) &temp.gnix_addr);
```

Comment:
In addendum, we can remove most of this logic in general by having the user specify a key length during init, so that we could just expect a specific key size for all entries and then the gnix_ht_key entry could be of any particular size, rather than limited to 64 bits.